### PR TITLE
BAEL-3009: Logging Spring WebClient calls

### PR DESCRIPTION
**Pull request by maryarm**
BAEL-3009:
Logging Spring WebClient calls.

### CODE REVIEW

File: spring-5-reactive-client/pom.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (rpvilao) [inline spring-5-reactive-client/pom.xml]: fix

File: spring-5-reactive-client/src/test/java/com/baeldung/reactive/logging/jetty/RequestLogEnhancer.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        request.onResponseContent((theResponse, content) -> group.append(toString(content, getCharset(theResponse.getHeaders()))));
+        request.onResponseSuccess(theResponse -> {
+            log.debug(group.toString());
+        });
+        return request;
+    }
+
+
+    private String toString(ByteBuffer buffer, Charset charset)
+    {
```
Comment (rpvilao) [inline spring-5-reactive-client/src/test/java/com/baeldung/reactive/logging/jetty/RequestLogEnhancer.java]: fix

File: spring-5-reactive-client/src/test/java/com/baeldung/reactive/logging/filters/LogFilters.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            if ( log.isDebugEnabled()) {
+                StringBuilder sb = new StringBuilder("Request: \n").append(clientRequest.method()).append(" ").append(clientRequest.url());
+                clientRequest
+                  .headers()
+                  .forEach((name, values) -> values.forEach(value -> sb.append("\n").append(name).append(":").append(value)));
+                log.debug(sb.toString());
+            }
+            return Mono.just(clientRequest);
+        }), ExchangeFilterFunction.ofResponseProcessor(clientResponse -> {
+            if ( log.isDebugEnabled()) {
```
Comment (rpvilao) [inline spring-5-reactive-client/src/test/java/com/baeldung/reactive/logging/filters/LogFilters.java]: formatting issue

File: spring-5-reactive-client/src/test/java/com/baeldung/reactive/logging/jetty/RequestLogEnhancer.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        byte[] bytes;
+        if (buffer.hasArray()) {
+            bytes = new byte[buffer.capacity()];
+            System.arraycopy(buffer.array(), 0, bytes, 0, buffer.capacity() );
+        }
+        else
+        {
```
Comment (rpvilao) [inline spring-5-reactive-client/src/test/java/com/baeldung/reactive/logging/jetty/RequestLogEnhancer.java]: fix

File: spring-5-reactive-client/src/test/java/com/baeldung/reactive/logging/netty/CustomLogger.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    protected String format(ChannelHandlerContext ctx, String eventName, Object arg) {
+        if (arg instanceof ByteBuf) {
+            ByteBuf msg = (ByteBuf) arg;
+            return decodeString(msg, msg.readerIndex(), msg.readableBytes(), Charset.defaultCharset());
+        }
+        return "";
+    }
+
+    private String decodeString(ByteBuf src, int readerIndex, int len, Charset charset) {
+        if (len == 0) return "";
```
Comment (rpvilao) [inline spring-5-reactive-client/src/test/java/com/baeldung/reactive/logging/netty/CustomLogger.java]: use brackets

File: spring-5-reactive-client/src/test/java/com/baeldung/reactive/logging/jetty/RequestLogEnhancer.java (hunk 2 above)
Comment (maryarm) [reply spring-5-reactive-client/src/test/java/com/baeldung/reactive/logging/jetty/RequestLogEnhancer.java]: fixed.

File: spring-5-reactive-client/src/test/java/com/baeldung/reactive/logging/filters/LogFilters.java (hunk 3 above)
Comment (maryarm) [reply spring-5-reactive-client/src/test/java/com/baeldung/reactive/logging/filters/LogFilters.java]: fixed!

File: spring-5-reactive-client/src/test/java/com/baeldung/reactive/logging/jetty/RequestLogEnhancer.java (hunk 4 above)
Comment (maryarm) [reply spring-5-reactive-client/src/test/java/com/baeldung/reactive/logging/jetty/RequestLogEnhancer.java]: fixed

File: spring-5-reactive-client/src/test/java/com/baeldung/reactive/logging/netty/CustomLogger.java (hunk 5 above)
Comment (maryarm) [reply spring-5-reactive-client/src/test/java/com/baeldung/reactive/logging/netty/CustomLogger.java]: fixed.

**Pull request merged**